notes/change-report-form-received.vbs

```diff
@@ -65,55 +70,72 @@ BeginDialog Dialog1, 0, 0, 376, 280, "Change Report Form Received"
+
+' Updated dialog to include field validation
 DO
 	DO
-		DO
-			DO
-    			Dialog Dialog1
-				cancel_confirmation
-				IF worker_signature = "" THEN MsgBox "You must sign your case note!"
-			LOOP UNTIL worker_signature <> ""
-			IF IsNumeric(MAXIS_case_number) = FALSE THEN MsgBox "You must type a valid numeric case number."
-		LOOP UNTIL IsNumeric(MAXIS_case_number) = TRUE
-		IF changes_continue = "Select One:" THEN MsgBox "You Must Select 'The changes client reports field'"
-	LOOP UNTIL changes_continue <> "Select One:"
-	call check_for_password(are_we_passworded_out)  'Adding functionality for MAXIS v.6 Passworded Out issue'
+		err_msg = ""
+		Dialog Dialog1
+		cancel_confirmation ' Asks user to confirm if they want to cancel
+
+		' Validate that MAXIS number is numeric and less than 8 digits long
+		CALL validate_MAXIS_case_number(err_msg, "* ")
+		' Validate that MAXIS case number is valid
+		CALL navigate_to_MAXIS_screen("CASE", "NOTE")
+		EMReadScreen invalid_case_number, 7, 24, 2
+		Call back_to_SELF()
+		If invalid_case_number = "INVALID" THEN err_msg = err_msg & vbNewLine & "* This is an invalid case number - update the case number."
+		' Validate that worker signature is not blank.
+		IF trim(worker_signature) = "" THEN err_msg = err_msg & vbCr & "* Please sign your case note."
+		' Validate that worker selects option from dropdown list as to how long change will last
+		If changes_continue = "Select One:" THEN err_msg = err_msg & vbNewLine & "* You must select an option from the dropdown list indicating whether the changes reported by the client will continue next month or will not continue next month."
+		' Validate the fields to ensure that at least one field is filled in besides the required fields (case number, worker signature, select one dropdown)
+		If date_received = "" AND effective_date = "" AND address_notes = "" AND household_notes = "" AND asset_notes = "" AND vehicles_notes = "" AND income_notes = "" AND shelter_notes = "" AND other_change_notes = "" AND actions_taken = "" AND other_notes = "" AND verifs_requested = "" THEN err_msg = err_msg & vbNewLine & "* All of the fields are blank. You must enter information in at least one field."
+		If err_msg <> "" THEN MsgBox "*** NOTICE!!! ***" & vbNewLine & err_msg
+	LOOP UNTIL err_msg = ""
+	CALL check_for_password(are_we_passworded_out)
 LOOP UNTIL are_we_passworded_out = false
 
-'Checks Maxis for password prompt
-CALL check_for_MAXIS(FALSE)
+
+
+' Check if case is privileged and end script if it is privileged
+Call navigate_to_MAXIS_screen_review_PRIV("CASE", "NOTE", is_this_priv)
+If is_this_priv = True then script_end_procedure("This case is privileged and cannot be accessed. The script will now stop.")
+' Confirm that the case is in county and end script if the case is out of county
+EMReadScreen county_code, 4, 21, 14
+If county_code <> worker_county_code then script_end_procedure("This case is out-of-county, and cannot access CASE:NOTE. The script will now stop.")
```

Comment:
Line 143: Again I know this isn't your code, but how about we have the effective date in the header of the case note?

Reply:
Added effective date to header of case note. I kept the formatting in the header as it appears in body of case note, but let me know if you had something else in mind for the header.